@navikt/core/react/src/form/virtualfocus/VirtualFocus.tsx

```diff
@@ -0,0 +1,228 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  forwardRef,
+  useId,
+  useState,
+} from "react";
+import { Slot } from "../../util/Slot";
+import { createContext } from "../../util/create-context";
+import { useMergeRefs } from "../../util/hooks";
+import { createDescendantContext } from "../../util/hooks/descendants/useDescendant";
+import { SlottedDivElementRef } from "./SlottedDivElement";
+
+export const [
+  VirtualFocusDescendantsProvider,
+  useVirtualFocusDescendantsContext,
+  useVirtualFocusDescendantInitializer,
+  useVirtualFocusDescendant,
+] = createDescendantContext<
+  SlottedDivElementRef,
+  {
+    handleOnSelect: () => void;
+    handleOnActive: () => void;
+  }
+>();
+
+const [VirtualFocusInternalContextProvider, useVirtualFocusInternalContext] =
+  createContext<{
+    virtualFocusIdx: number;
+    setVirtualFocusIdx: Dispatch<SetStateAction<number>>;
+    loop: boolean;
+    uniqueId: string;
+  }>();
+
+type VirtualFocusProps = {
+  children: React.ReactNode;
+  /**
+   * Whether to cause focus to loop around when it hits the first or last element
+   * @default true
+   **/
+  loop?: boolean;
+};
+
+export const VirtualFocus = ({ children, loop = false }: VirtualFocusProps) => {
+  const descendants = useVirtualFocusDescendantInitializer();
+  const [virtualFocusIdx, setVirtualFocusIdx] = useState(0);
+
+  return (
+    <VirtualFocusInternalContextProvider
+      virtualFocusIdx={virtualFocusIdx}
+      setVirtualFocusIdx={setVirtualFocusIdx}
+      loop={loop}
+      uniqueId={useId().replace(/:/g, "")}
```

Comment:
Since we still support React v17, we can't use React built-in `useId`(React 18), have to use this instead
```
import { useId } from "../../util/hooks";
```

Reply:
Can also remove `.replace(/:/g, "")` then i think